### PR TITLE
Fix localization on input for FiatAmountInputCard

### PR DESCRIPTION
**Pull request by Jon-edge**
#### PR Dependencies

 none

#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [ ] Tested on iOS device
- [ ] Tested on Android device
- [ ] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203475087427001

### CODE REVIEW

File: src/components/cards/FiatAmountInputCard.tsx [hunk 1]
```diff
@@ -41,29 +41,33 @@ const FiatAmountInputCardComponent = ({ wallet, iconUri, inputModalMessage, titl
...
+      destToFiatRate == null || destToFiatRate === '0'
+        ? '0'
+        : mul(destExchangeMultiplier, div(fiatAmount ?? '0', destToFiatRate, DECIMAL_PRECISION)),
       0
     )
 
     setNativeCryptoAmount(calculatedNativeCryptoAmount)
-    if (!zeroString(calculatedNativeCryptoAmount) && !zeroString(fiatAmount)) onAmountChanged(fiatAmount, calculatedNativeCryptoAmount)
-  }, [allTokens, destToFiatRate, fiatAmount, onAmountChanged, tokenId, wallet.currencyInfo])
+    if (!zeroString(calculatedNativeCryptoAmount) && !zeroString(inputFiatAmount)) onAmountChanged(inputFiatAmount, calculatedNativeCryptoAmount)
```
Comment (peachbits) [inline src/components/cards/FiatAmountInputCard.tsx:55]: This should be reverted so the (now) sanitized value is passed to `zeroString` and `onAmountChanged`. `zeroString` uses biggystring directly as well one of the parents of FiatAmountInputCard passes an `onAmountChanged` that eventually leads to biggystring.

File: src/components/cards/FiatAmountInputCard.tsx [hunk 2]
```diff
@@ -41,29 +41,33 @@ const FiatAmountInputCardComponent = ({ wallet, iconUri, inputModalMessage, titl
...
+  }, [allTokens, destToFiatRate, inputFiatAmount, onAmountChanged, tokenId, wallet.currencyInfo])
 
   const handleEditActionfiatAmount = React.useCallback(() => {
     Airship.show<string | undefined>(bridge => <TextInputModal title={title} message={inputModalMessage} bridge={bridge} keyboardType="decimal-pad" />).then(
       inputAmount => {
         if (inputAmount != null && !zeroString(inputAmount)) {
-          setFiatAmount(inputAmount)
-          onAmountChanged(fiatAmount, nativeCryptoAmount)
+          setInputFiatAmount(inputAmount)
+          onAmountChanged(inputFiatAmount, nativeCryptoAmount)
```
Comment (peachbits) [inline src/components/cards/FiatAmountInputCard.tsx:63]: Same issue here

**Review by peachbits — CHANGES_REQUESTED** [review]

**Comment by Jon-edge** [comment]
Closing to open https://github.com/EdgeApp/edge-react-gui/pull/3838 with requested changes and a fixed branch name

**Comment by Jon-edge** [comment]
Closing to open https://github.com/EdgeApp/edge-react-gui/pull/3838 with requested changes and a fixed branch name

**Pull request closed without merging**